interface/orders/receive_hl7_results.inc.php

```diff
@@ -469,13 +464,9 @@ function match_patient($ptarr)
     } else {
         // No match good enough, figure out if there's enough ambiguity to ask the user.
         $tmp = sqlQuery(
-            "SELECT pid FROM patient_data WHERE " .
-            "(ss IS NOT NULL AND ss != '' AND REPLACE(ss, '-', '') = ?) OR " .
-            "(fname IS NOT NULL AND fname != '' AND fname = ? AND " .
-            "lname IS NOT NULL AND lname != '' AND lname = ?) OR " .
-            "(DOB IS NOT NULL AND DOB = ?) " .
-            "LIMIT 1",
-            array($in_ss, $in_fname, $in_lname, $in_dob)
+            "SELECT pid FROM patient_data WHERE MATCH(lname) " .
+            " AGAINST(? IN BOOLEAN MODE) AND fname = ? AND DOB = ? AND sex = ?",
+            array($in_lname, $in_fname, $in_dob, $in_sex)
```

Comment:
this is the same query as above, note how the earlier query looks for other potential matches that the user could be presented with

Reply:
I will make an adjustment to the query. The previous queries were very similar but slightly different. I have noted the difference between the two. I am testing today and will post an update by the end of the week.

Reply:
What I tried did not work. I will review the workflow of the code some more. The only thing I can see (in my limited view) is to add ss to the second query. But in our experience right now. We are getting a 99% accuracy with the new query that the second query is not needed to do what the original author intended. I will capture some data to see if the second query is run at all.